tests/csapi/apidoc_content_test.go

```diff
@@ -0,0 +1,63 @@
+package csapi_tests
+
+import (
+	"io/ioutil"
+	"net/url"
+	"strings"
+	"testing"
+
+	"github.com/tidwall/gjson"
+
+	"github.com/matrix-org/complement/internal/b"
+	"github.com/matrix-org/complement/internal/client"
+)
+
+var contentID *url.URL
+var contentText = "Here is the content I am uploading"
+
+func TestContent(t *testing.T) {
+	deployment := Deploy(t, b.BlueprintAlice)
+	defer deployment.Destroy(t)
+	authedClient := deployment.Client(t, "hs1", "@alice:hs1")
+	unauthedClient := deployment.Client(t, "hs1", "")
+	// sytest: POST /media/r0/upload can create an upload
+	t.Run("POST /media/r0/upload can create an upload", func(t *testing.T) {
+
+		gotUrl := ""
+		queryParams := url.Values{}
+		queryParams.Set("access_token", authedClient.AccessToken)
+
+		contentType := client.WithContentType("text/plain")
+		reqBody := client.WithRawBody([]byte(contentText))
+		res := unauthedClient.MustDoFunc(t, "POST", []string{"_matrix", "media", "r0", "upload"}, reqBody, contentType, client.WithQueries(queryParams))
+
+		body, err := ioutil.ReadAll(res.Body)
+		if err != nil {
+			t.Fatalf("unable to read response body: %v", err)
+		}
+
+		gotUrl = gjson.GetBytes(body, "content_uri").Str
+		if gotUrl == "" {
+			t.Fatal("expected URL to hae propogated the value")
+		}
+
+		contentID, err = url.Parse(gotUrl)
+		if err != nil {
+			t.Fatal("invalid URL returned in response")
```

Comment:
Please log the error and the `gotUrl`.

It is never sufficient to just log a simple error message in tests like this. Think what the experience is like for someone whose code is failing because of this test. What information would they find useful? They certainly will want to know _why_ the test is failing (the `err` in this case). Giving them some more context to try to help them debug this (the `gotUrl` in this case) is also helpful.

Think about it like when you write tests in Complement. Sometimes the server encounters an error, and Complement gives detailed error messages why. It doesn't just say "request failed" because it just isn't helpful.